### PR TITLE
GH-1280: As a user I want to use ECMAScript imports (with ESM, webpack) as experimental feature

### PR DESCRIPTION
**Pull request by mor-n4**
See #1280.

This PR does not include the changes required in n4js-libs (mostly `n4js-node`) as those won't be merged to master as part of GH-1280; instead, the changes in n4js-libs will remain on a branch for the time being and will be published under a special dist-tag.

### CODE REVIEW

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/internal/N4JSModel.java [hunk 1]
```diff
@@ -442,6 +442,17 @@ public String getDefinesPackage(final IN4JSProject project) {
 		return null;
 	}
 
+	public boolean isUseES6Imports(final IN4JSProject project) {
+		if (project == null) {
+			return false;
+		}
+		final ProjectDescription projectDescription = getProjectDescription(project.getLocation());
+		if (projectDescription != null) {
```
Comment (qtran-n4) [inline plugins/org.eclipse.n4js/src/org/eclipse/n4js/internal/N4JSModel.java]: minor but to be consistent 

```
if (projectDescription == null) {
   return null;
}
```

?

File: plugins/org.eclipse.n4js.json.model/src/org/eclipse/n4js/json/model/utils/JSONModelUtils.java [hunk 2]
```diff
@@ -52,17 +53,20 @@
 	/** The JSON file extension. */
 	public static final String FILE_EXTENSION = "json";
 
+	/**
+	 * If the given JSON value is a {@link JSONBooleanLiteral} with a value of <code>true</code>, returns
+	 * <code>true</code>, otherwise <code>false</code>.
+	 */
+	public static boolean asBoolean(JSONValue jsonValue) {
```
Comment (qtran-n4) [inline plugins/org.eclipse.n4js.json.model/src/org/eclipse/n4js/json/model/utils/JSONModelUtils.java]: `asBooleanOrFalse` to be consistent with `asStringOrNull` below?

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/internal/N4JSModel.java (hunk 1 above)
Comment (mor-n4) [reply plugins/org.eclipse.n4js/src/org/eclipse/n4js/internal/N4JSModel.java]: I had aligned that to the method right above this one.

File: plugins/org.eclipse.n4js.json.model/src/org/eclipse/n4js/json/model/utils/JSONModelUtils.java (hunk 2 above)
Comment (mor-n4) [reply plugins/org.eclipse.n4js.json.model/src/org/eclipse/n4js/json/model/utils/JSONModelUtils.java]: done

**Comment by mor-n4** [comment]
Thanks! Merging ...

**Pull request merged**